packages/transit-vehicle-overlay/src/utils.js

```diff
@@ -0,0 +1,298 @@
+  }
+  return retVal;
+}
+
+/**
+ * find a vehicle based on an id (either vehicle or trip id) within a list of vehicles
+ * @param vehicleList - list of vehciles to be searched
+ * @param queryId - either trip or vehcile id (string) to search list
+ * @param defVal - return value when not found or errors happen
+ */
+export function findVehicleById(vehicleList, queryId, defVal = null) {
+  let retVal = defVal;
+  try {
+    if (vehicleList && queryId) {
+      vehicleList.some(v => {
+        if (queryId === v.vehicleId || queryId === v.tripId) {
+          retVal = v;
+          return true;
+        }
+        return false;
+      });
+    }
+  } catch (e) {
+    console.log(e);
+  }
+  return retVal;
+}
+
+/**
+ * do 2 vehicle have the same tracking ids
+ * @param vehicleA
+ * @param vehicleB
+ * @returns boolean
+ */
+export function isTracked(vehicleA, vehicleB) {
+  let retVal = false;
+
+  try {
+    if (vehicleA && vehicleB) {
+      if (
+        vehicleA.vehicleId === vehicleB.vehicleId ||
+        vehicleA.tripId === vehicleB.tripId
+      ) {
+        retVal = true;
+      }
+    }
+  } catch (e) {
+    console.log(e);
+  }
+  return retVal;
+}
+
+/** build a url from sub parts */
+export function buildUrl(base, query) {
+  let retVal = base;
+  try {
+    retVal = `${base}/${query}`.replace(/([^:]\/)\/+/g, "$1");
+  } catch (e) {
+    retVal = `${base}/${query}`;
+  }
+  return retVal;
+}
+
+/**
+ * query real-time vehicles, and return the results in the set* parameter callbacks
+ *
+ * @param setData - callback where vehicles array will be passed
+ * @param url - url of the vehicle service
+ */
+export function fetchVehicles(setData, trackedId, baseUrl, query) {
+  // build url
+  baseUrl = baseUrl || "https://maps.trimet.org/gtfs/rt/vehicles/";
+  query = query || "routes/all";
+  let url = buildUrl(baseUrl, query);
+
+  // append date to end of url (caching, etc...)
+  const d = Date.now();
+  url = url.indexOf("?") ? `${url}?` : `${url}&`;
+  url = `${url}__time__=${d}`;
+
+  fetch(url)
+    .then(res => {
+      if (!res.ok) {
+        console.log(res.statusText);
+        throw Error(res.statusText);
+      }
+      return res;
+    })
+    .then(res => {
+      const retVal = res.json();
+      return retVal;
+    })
+    .then(vehicleList => {
+      if (vehicleList && vehicleList.length > 0) {
+        // step 1: set the vehicle list (triggers vehicle points redraw)
+        console.log(`updating state with ${vehicleList.length} vehicles`);
+        setData(vehicleList, trackedId);
+      } else {
+        console.log("get vehicle data is suspect");
+      }
+    })
+    .catch(error => {
+      console.log(`VEH fetch() error: ${error} for ${url}`);
+    });
+}
+
+/**
+ * fetch line geometry for a given pattern
+ *
+ * https://newplanner.trimet.org/ws/ti/v0/index/patterns/TRIMET:433758/geometry/geojson
+ * https://newplanner.trimet.org/ws/ti/v0/index/patterns/{agency}:{pattern}/geometry/geojson
```

Comment:
Some more context might be able to be added to note that these URLs are examples to look at.